server/authentication/passport.ts

```diff
@@ -97,26 +97,34 @@ export default function passportSetup(app: Application, hmppsAuthService: HmppsA
   app.get('/sign-in', signIn)
 
   const signInCallback = (redirectUri: string) =>
+    // TODO - remove ts-ignore once keepSessionInfo has been added to types/passport
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore - No overload matches this call. - keep this until @types/passport updated
     passport.authenticate('oauth2', {
       successReturnToOrRedirect: '/',
       failureRedirect: redirectUri,
+      keepSessionInfo: true,
     })
   app.get('/sign-in/callback', signInCallback('/sign-in'))
 
-  const signOut = (logoutTerm: string, req: Request, res: Response) => {
+  const signOut = (logoutTerm: string, req: Request, res: Response, next: NextFunction) => {
     const { redirectUrl } = req.query
     const logoutUrl = `${authConfig.url}/${logoutTerm}?client_id=${authConfig.loginClientId}&redirect_uri=${
       config.domain
     }${redirectUrl || `/${logoutTerm}/success`}`
     if (req.isAuthenticated()) {
-      req.logout()
-      req.session!.destroy(() => res.redirect(logoutUrl))
```

Comment:
change to add callback function (copied from [example pattern](https://www.passportjs.org/tutorials/auth0/logout/))

note `req.session.destroy` removed as `unset` option added to session in `app.ts`